htlcswitch/link.go

```diff
@@ -2382,6 +2423,27 @@ func (l *channelLink) handleUpstreamMsg(msg lnwire.Message) {
 
 }
 
+// blindedRouteFailure looks up the originally added htlc, and provides a
+// switched out failure message if it was part of a blinded route.
+func (l *channelLink) blindedRouteFailure(htlcIdx uint64) (
+	lnwire.FailureMessage, error) {
```

Comment:
should we do "lnwire.FailureMessage, bool, error" just to stick to the "if a bool is returned, check that before referencing" pattern? just to avoid accidental nil pointer dereferences in future.